include/zephyr/bluetooth/audio/mcc.h

```diff
@@ -611,6 +672,7 @@ int bt_mcc_read_player_name(struct bt_conn *conn);
 int bt_mcc_read_icon_obj_id(struct bt_conn *conn);
 #endif /* CONFIG_BT_OTS_CLIENT */
 
+#if defined(CONFIG_BT_MCC_READ_MEDIA_PLAYER_ICON_URL)
```

Comment:
Ditto there is no reason to guard the definitions of the functions; just the implementations of them.